concert/experiments/imaging.py

```diff
@@ -279,7 +284,13 @@ async def _produce_frames(self, number, **kwargs):
         """
         await self._camera.set_trigger_source("AUTO")
         async with self._camera.recording():
+            if isinstance(self, Tomography):
+                self.log.info("First frame started at position %s" %
+                              await self._tomography_motor.get_position())
+
             for i in range(int(number)):
+                if self._stop:
```

Comment:
Also, instead of checking this in _every_ experiment class, we can patch catch a stop request in the `Acquistion`, namely you can wrap the `broadcast` into something which accepts a stop request.

Reply:
But what will it change? I have to write everywhere, where right now is `if self._stop` a handler of the stop request?

Reply:
But a clean stop is very experiment specific. So I abort all the acquisitions and then call a clean-up function, that has to take care that everything is back in a well defined state?

Reply:
In all your producers you just do `if stop then break`, so no clean up. And of course if special handler is needed you can always override the `Experiment.stop` method.